src/components/Api.js

```diff
@@ -0,0 +1,93 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  foundResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}, ${error.message}`);
+  }
+
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers
+    }).then((res) => this.foundResponse(res));
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers
+    }).then((res) => this.foundResponse(res));
+  }
+
+  getUserAndCards() {
+    return Promise.all([this.getUserInfo(), this.getInitialCards()])
+      .then(([userInfo, cards]) => {
+        return { userInfo, cards };
+      })
+      .catch((err) => {
+        console.error('Error in getUserAndCards:', err);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.